injector/src/main/java/edu/ucr/cs/riple/injector/Helper.java

```diff
@@ -210,6 +219,9 @@ private static Node findTopLevelAnonymousClass(Node cursor, int index) {
           }
           return false;
         });
+    if (index >= candidates.size()) {
+      throw new TargetClassNotFound("Top-Level-Anonymous-Class", "$" + index, cursor);
+    }
```

Comment:
This might not be quite as easy as this, though... what happens if there are `Foo$1`, `Foo$2`, and `Foo$3`, but it is `Foo$2` that is generated? We might some additional information from NullAway to handle this properly in the case of anonymous classes specifically (Fortunately, Lombok generated _anonymous_ classes is not something I have run into, at least outside of Lombok-generated methods. Still, I wonder if we should fail loudly in that case, rather than silently pick the wrong anonymous inner class)

Reply:
Either way: worth tracking as a future task somewhere. Not high priority, but definitely a potential pitfall/bug.

Reply:
Sure, that's a good point. Will create an issue once this lands, to track it.